### PR TITLE
[CDAP-12236, 12337] Implements condition node in UI

### PR DESCRIPTION
**Pull request by tonybach**
JIRAs:
- https://issues.cask.co/browse/CDAP-12236
- https://issues.cask.co/browse/CDAP-12237

Currently only the stage styling is done, waiting for the backend to provide the plugin so we can work on the configuration.

### CODE REVIEW

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js [hunk 1]
```diff
@@ -258,23 +259,61 @@ angular.module(PKG.name + '.commons')
...
+          }
+          let newTrueEndpoint = vm.instance.addEndpoint(node.name, {
+            anchor: 'Right',
+            cssClass: 'condition-endpoint condition-endpoint-true',
+            isSource: true,
+            endpoint: 'Dot',
+            uuid: 'True' + node.name,
+            connectorStyle: vm.conditionTrueConnectionStyle,
+            overlays: [
+              [ 'Label', { label:'Yes', id:'yesLabel', location:[0.5, -0.55], cssClass: 'condition-label' } ]
```
Comment (elfenheart) [inline cdap-ui/app/directives/dag-plus/my-dag-ctrl.js]: nit: space after `:`

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js [hunk 2]
```diff
@@ -258,23 +259,61 @@ angular.module(PKG.name + '.commons')
...
+
+          let newFalseEndpoint = vm.instance.addEndpoint(node.name, {
+            anchor: 'Bottom',
+            cssClass: 'condition-endpoint condition-endpoint-false',
+            isSource: true,
+            endpoint: 'Dot',
+            uuid: 'False' + node.name,
+            connectorStyle: vm.conditionFalseConnectionStyle,
+            overlays: [
+              [ 'Label', { label:'No', id:'noLabel', location:[0.5, -0.55], cssClass: 'condition-label' } ]
```
Comment (elfenheart) [inline cdap-ui/app/directives/dag-plus/my-dag-ctrl.js]: here too

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js [hunk 3]
```diff
@@ -423,23 +519,39 @@ angular.module(PKG.name + '.commons')
...
+              } else if (sourceEndpoint.canvas.classList.contains('condition-endpoint-false')) {
+                connObj.connection.setType('conditionFalse');
+              }
+            }
+          } else {
+            connObj.connection.setType('basic');
+          }
+          if (targetNode.type === 'condition') {
+            connObj.connection.addType('dashed');
+          }
```
Comment (elfenheart) [inline cdap-ui/app/directives/dag-plus/my-dag-ctrl.js:557]: we might have to do dashed connection for connection from/to `Action` as well?

File: cdap-ui/package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
   },
   "devDependencies": {
     "autoprefixer": "6.5.0",
-    "babel-core": "https://registry.npmjs.org/babel-core/-/babel-core-6.9.1.tgz",
+    "babel-core": "6.26.0",
```
Comment (elfenheart) [inline cdap-ui/package.json]: remove this change

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js (hunk 3 above)
Comment (tonybach) [reply cdap-ui/app/directives/dag-plus/my-dag-ctrl.js:557]: Yep, added

**Comment by bdmogal** [comment]
@tonybach let's group Conditions with Actions in the palette and give the accordion tab the header `Conditions and Actions`

**Comment by tonybach** [comment]
Also added a commit for https://issues.cask.co/browse/CDAP-12397 to change `Configure Properties` and `View Properties` labels to just `Properties`.

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Comment by leacd** [comment]
LGTM

**Pull request merged**